connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,243 @@
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+  /v1/streams/list:
+    post:
+      summary: List all streams present in the connector definition, along with their specific request URLs
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StreamsListRequestBody"
+        required: true
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamsListRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+
+components:
+  schemas:
+    StreamRead:
+      type: object
+      required:
+        - slices
+      properties:
+        slices:
+          type: array
+          description: The stream slices returned from the read command
+          items:
+            type: object
+            required:
+              - sliceDescriptor
+              - pages
+            properties:
+              sliceDescriptor:
+                type: object
+                description: 'An object describing the current slice, e.g. {start_time: "2021-01-01", end_time: "2021-01-31"}'
+              pages:
+                type: array
+                description: The pages returned from the read command
+                items:
+                  type: object
+                  required:
+                    - airbyteMessages
+                    - request
+                    - response
+                  properties:
+                    airbyteMessages:
+                      type: array
+                      description: The RECORD/STATE/LOG AirbyteMessages coming from the read operation for this page
+                      items:
+                        $ref: "#/components/schemas/AirbyteProtocol/definitions/AirbyteMessage"
+                    request:
+                      $ref: "#/components/schemas/HttpRequest"
+                    response:
+                      $ref: "#/components/schemas/HttpResponse"
+    StreamReadRequestBody:
+      type: object
+      required:
+        - definition
+        - stream
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+        stream:
+          type: string
+          description: Name of the stream to read
+        numPages:
+          type: integer
+          description: Number of pages to read from the source for each slice
+          default: 1
+        state:
```

Comment:
should this be a proper STATE message?

Reply:
Good callout! Yes it should be, will update